packages/web-app-files/tests/unit/views/LocationPicker.spec.js

```diff
@@ -0,0 +1,459 @@
+      action => {
+        const wrapper = getShallowWrapper({ $route: getRoute({ action: action }) })
+
+        expect(wrapper).toMatchSnapshot()
+      }
+    )
+    describe('should display breadcrumb with items', () => {
+      it.each(['parent/child', 'parent//child', '/parent/child', '/parent/child/'])(
+        'if route params context is public',
+        input => {
+          const route = getRoute({ context: 'public', item: input })
+          const wrapper = getShallowWrapper({ $route: route })
+          const breadcrumbItems = wrapper.find(breadcrumbStub).props().items
+
+          expect(breadcrumbItems.length).toBe(2)
+          expect(breadcrumbItems[0]).toMatchObject({
+            index: 0,
+            text: 'Public link',
+            to: {
+              name: route.name,
+              params: {
+                action: route.params.action,
+                item: 'parent'
+              },
+              query: {
+                resource: [route.query.resource]
+              }
+            }
+          })
+          expect(breadcrumbItems[1]).toMatchObject({
+            index: 2,
+            text: 'child'
+          })
+          expect(breadcrumbItems[1].to).toBeUndefined()
+        }
+      )
+      it.each(['parent/child', 'parent//child', '/parent/child', '/parent/child/'])(
+        'if route params context is not public',
+        input => {
+          const route = getRoute({ item: input })
+          const wrapper = getShallowWrapper({ $route: route })
+          const breadcrumbItems = wrapper.find(breadcrumbStub).props().items
+
+          expect(breadcrumbItems.length).toBe(3)
+          expect(breadcrumbItems[0]).toMatchObject({
+            index: 0,
+            text: 'All files',
+            to: {
+              name: route.name,
+              params: { action: route.params.action, item: '/' },
+              query: {
+                resource: [route.query.resource]
+              }
+            }
+          })
+          expect(breadcrumbItems[1]).toMatchObject({
+            index: 1,
+            text: 'parent',
+            to: {
+              name: route.name,
+              params: { action: route.params.action, item: 'parent' },
+              query: {
+                resource: [route.query.resource]
+              }
+            }
+          })
+          expect(breadcrumbItems[2]).toMatchObject({
+            index: 2,
+            text: 'child'
+          })
+          expect(breadcrumbItems[2].to).toBeUndefined()
+        }
+      )
+    })
+    describe('cancel button', () => {
+      it('should show the location picker cancel button', () => {
+        const wrapper = getShallowWrapper()
+        const cancelButton = wrapper.find(selectors.cancelButton)
+
+        expect(cancelButton.exists()).toBeTruthy()
+        expect(cancelButton.find(translateStub).exists()).toBeTruthy()
+        expect(cancelButton).toMatchSnapshot()
```

Comment:
Since this is ODS component I do not see a real reason to check the snapshot here...